src/services/createRouter.ts

```diff
@@ -72,17 +70,17 @@ export function createRouter<
   const TRoutes extends Routes,
   const TOptions extends RouterOptions,
   const TPlugin extends RouterPlugin = EmptyRouterPlugin
->(arrayOfRoutes: TRoutes[], options?: TOptions, plugins?: TPlugin[]): Router<TRoutes, TOptions, TPlugin>
+>(routes: TRoutes[], options?: TOptions, plugins?: TPlugin[]): Router<TRoutes, TOptions, TPlugin>
 
 export function createRouter<
   const TRoutes extends Routes,
   const TOptions extends RouterOptions,
   const TPlugin extends RouterPlugin = EmptyRouterPlugin
 >(routesOrArrayOfRoutes: TRoutes | TRoutes[], options?: TOptions, plugins: TPlugin[] = []): Router<TRoutes, TOptions, TPlugin> {
-  const pluginRoutes = plugins.map((plugin) => plugin.routes)
-  const routes = insertBaseRoute([...routesOrArrayOfRoutes, ...pluginRoutes].flat(), options?.base)
+  const routes = getRoutesForRouter(routesOrArrayOfRoutes, plugins, options?.base)
+  const hooks = createRouterHooks()
 
-  checkDuplicateNames(routes)
+  hooks.addRouteHooks(getGlobalHooksForRouter(options, plugins))
```

Comment:
I know you wanted something besides the router to add the global hooks. But it didn't really feel good passing the global routes into `createRouterHooks` (even though that's what I started with). I think the router IS responsible for registering its global hooks. But we can abstract away the logic for collecting them all (in the `getGlobalHooksForRouter` utility)